examples/xgboost_cv.py

```diff
@@ -0,0 +1,109 @@
+(1) Execute this code directly.
+    $ python xgboost_cv.py
+(2) Execute through CLI.
+    $ STUDY_NAME=`optuna create-study --direction maximize --storage sqlite:///example.db`
+    $ optuna study optimize xgboost_cv.py objective --n-trials=100 --study $STUDY_NAME \
+      --storage sqlite:///example.db
+"""
+
+
+import optuna
+from optuna.samplers import TPESampler
+import os
+import sklearn.datasets
+import sklearn.metrics
+import xgboost as xgb
+
+
+# Set path to save CV results
+file_path = "./tmp_results_xgbCV_optuna/"
+
+# Create folder if it doesnt exist
+if not os.path.exists(file_path):
+    try:
+        os.makedirs(file_path)
+    except Exception:
+        print("please create folder to store xgboost CV results")
+
+# Set Constants
+seed = 108
+n_folds = 3
+
+# Make the Optuna Sampler behave in a deterministically.
+sampler = TPESampler(seed=seed)
+
+
+# Define Optuna Objective
+# FYI: Objective functions can take additional arguments
+# (https://optuna.readthedocs.io/en/stable/faq.html#objective-func-additional-args).
+def objective(trial):
+    (data, target) = sklearn.datasets.load_breast_cancer(return_X_y=True)
+    dtrain = xgb.DMatrix(data, label=target)
+
+    param = {
+        "silent": 1,
+        "objective": "binary:logistic",
+        "eval_metric": "auc",
+        "booster": trial.suggest_categorical("booster", ["gbtree"]),
+        "alpha": trial.suggest_loguniform("alpha", 1e-3, 1.0),
+    }
+
+    if param["booster"] == "gbtree" or param["booster"] == "dart":
+        param["max_depth"] = trial.suggest_int("max_depth", 1, 9)
+        param["min_child_weight"] = trial.suggest_int("min_child_weight", 1, 9)
+        param["eta"] = trial.suggest_loguniform("eta", 1e-3, 1.0)
+        param["gamma"] = trial.suggest_loguniform("gamma", 1e-3, 1.0)
+        param["subsample"] = trial.suggest_loguniform("subsample", 0.6, 1.0)
+        param["colsample_bytree"] = trial.suggest_loguniform("colsample_bytree", 0.6, 1.0)
+        param["grow_policy"] = trial.suggest_categorical("grow_policy", ["depthwise", "lossguide"])
+
+    if param["booster"] == "dart":
+        param["sample_type"] = trial.suggest_categorical("sample_type", ["uniform", "weighted"])
+        param["normalize_type"] = trial.suggest_categorical("normalize_type", ["tree", "forest"])
+        param["rate_drop"] = trial.suggest_loguniform("rate_drop", 1e-8, 1.0)
+        param["skip_drop"] = trial.suggest_loguniform("skip_drop", 1e-8, 1.0)
+
+    xgb_cv_results = xgb.cv(
+        params=param,
+        dtrain=dtrain,
+        num_boost_round=10000,
+        nfold=n_folds,
+        stratified=True,
+        early_stopping_rounds=100,
+        seed=seed,
+        verbose_eval=False,
+    )
+
+    # (Optional)
+    # Print n_estimators in the output at each call to the objective function
+    print(
+        "-" * 10,
+        "Trial {} has optimal trees: {}".format(trial.number, str(xgb_cv_results.shape[0])),
+        "-" * 10,
+    )
+    print()
```

Comment:
I think this print statements make the output a bit verbose. How about removing it and showing `n_estimators` at the end of the execution (i.e., L109)?

```console
---------- Trial 8 has optimal trees: 1696 ----------

[I 2020-05-15 07:17:48,823] Finished trial#8 with value: 0.9882563333333333 with parameters: {'booster': 'gbtree', 'alpha': 0.11006970160449274, 'max_depth': 8, 'min_child_weight': 7, 'eta': 0.0016253741410996155, 'gamma': 0.019000120588029624, 'subsample': 0.6744265008652889, 'colsample_bytree': 0.6080708165519185, 'grow_policy': 'lossguide'}. Best is trial#7 with value: 0.991965.
---------- Trial 9 has optimal trees: 1408 ----------

[I 2020-05-15 07:17:51,772] Finished trial#9 with value: 0.9897633333333333 with parameters: {'booster': 'gbtree', 'alpha': 0.0027852366728488695, 'max_depth': 8, 'min_child_weight': 7, 'eta': 0.002435148116311601, 'gamma': 0.019866111168368436, 'subsample': 0.8959344463932998, 'colsample_bytree': 0.7192059963807262, 'grow_policy': 'depthwise'}. Best is trial#7 with value: 0.991965.

```

Reply:
Removed.